### DISCUSSION TITLE
Notifications sending to the wrong email address and can't get Dropbox to work

### DISCUSSION THREAD

**Discussion by matthewknill** [spatie/laravel-backup#1562] [Q&A]
**Version Info**
PHP Version: 8.0
Laravel: 8.83
spatie/laravel-backup: 7.8.0

**First Issue**
I have set the following in the config:
```php
'notifications' => [

        'notifications' => [
            // Failed
            \Spatie\Backup\Notifications\Notifications\BackupHasFailedNotification::class => ['mail'],
            \Spatie\Backup\Notifications\Notifications\UnhealthyBackupWasFoundNotification::class => ['mail'],
            \Spatie\Backup\Notifications\Notifications\CleanupHasFailedNotification::class => ['mail'],

            // Success
            \Spatie\Backup\Notifications\Notifications\BackupWasSuccessfulNotification::class => [],
            \Spatie\Backup\Notifications\Notifications\HealthyBackupWasFoundNotification::class => [],
            \Spatie\Backup\Notifications\Notifications\CleanupWasSuccessfulNotification::class => [],
        ],

        /*
         * Here you can specify the notifiable to which the notifications should be sent. The default
         * notifiable will use the variables specified in this config file.
         */
        'notifiable' => \Spatie\Backup\Notifications\Notifiable::class,

        'mail' => [
            'to' => env('MAIL_ERROR_TO_ADDRESS'),

            'from' => [
                'address' => env('MAIL_FROM_ADDRESS', 'hello@example.com'),
                'name' => env('MAIL_FROM_NAME', 'Example'),
            ],
        ],
```

Then in my `.env` file, I have set the following:
``
MAIL_ERROR_TO_ADDRESS=myemailaddress@example.com
``

However, it seems that errors are sent to the email address associated with the Dropbox account that is used for backups...

**Second Issue**
I'm also finding it very difficult to get Dropbox working and am constantly getting `401 Unauthorized` errors. Dropbox now uses tokens that expire after about 4 hours and I'm unsure how to get the refresh working with the backup (run by cron). I tried making them not expire with [this](https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/Tokens-only-valid-for-4-hours-from-app-console/m-p/425358/highlight/true#M22718) but the issue seems to persist.

**Comment by angeljqv** [comment]
> I'm also finding it very difficult to get Dropbox working and am constantly getting `401 Unauthorized` errors.

You must ask that on Dropbox Package, nobody here knows that plugin, maybe a replace [flysystem-google-drive-ext](https://github.com/masbug/flysystem-google-drive-ext)

**Comment by angeljqv** [comment]
> Notifications sending to the wrong email address

Did you clean config cache??
Feel free to do a PR to fix that or a failed test to show the problem

**Comment by matthewknill** [comment]
> > I'm also finding it very difficult to get Dropbox working and am constantly getting `401 Unauthorized` errors.
> 
> You must ask that on Dropbox Package, nobody here knows that plugin, maybe a replace [flysystem-google-drive-ext](https://github.com/masbug/flysystem-google-drive-ext)

The link that you sent seems to be only related to Google Drive and not Dropbox...

> > Notifications sending to the wrong email address
> 
> Did you clean config cache?? Feel free to do a PR to fix that or a failed test to show the problem

Yes cache was cleared and I can confirm that the notification is coming from Spatie Backup:
![image](https://user-images.githubusercontent.com/50228686/182969223-b12efd66-3f4b-4933-9568-11ae7946bbe6.png)

**Comment by angeljqv** [comment]
> Dropbox now uses tokens that expire after about 4 hours

That is problem on dropbox package, not on this package

**Reply by matthewknill** [reply]
Would you happen to know what package it is so I can ask a question on there?

**Reply by angeljqv** [reply]
Nobody can tell what packages are you using, only you can know that
`composer show -i`
